src/envoy/mixer/mixer_control.cc

```diff
@@ -74,13 +77,29 @@ HttpMixerControl::HttpMixerControl(const HttpMixerConfig& mixer_config,
 TcpMixerControl::TcpMixerControl(const TcpMixerConfig& mixer_config,
                                  Upstream::ClusterManager& cm,
                                  Event::Dispatcher& dispatcher,
-                                 Runtime::RandomGenerator& random) {
+                                 Runtime::RandomGenerator& random)
+    : dispatcher_(dispatcher) {
   ::istio::mixer_control::tcp::Controller::Options options(
       mixer_config.tcp_config);
 
   CreateEnvironment(cm, dispatcher, random, &options.env);
 
   controller_ = ::istio::mixer_control::tcp::Controller::Create(options);
+
+  if (mixer_config.tcp_config.has_report_interval() &&
+      mixer_config.tcp_config.report_interval().seconds() >= 0 &&
+      mixer_config.tcp_config.report_interval().nanos() >= 0) {
+    report_interval_ms_ = std::chrono::milliseconds(
+        mixer_config.tcp_config.report_interval().seconds() * 1000 +
+        mixer_config.tcp_config.report_interval().nanos() / 1000000);
+    // If configured time interval is less than 1 millisecond, then set report
+    // interval to default value.
+    if (report_interval_ms_ == std::chrono::milliseconds::zero()) {
```

Comment:
We may not want to sent more frequent than 0.5 second per report, or even 1 second.   0.5 is good for testing.